### PR TITLE
New sys_invocation_state and sys_journal tables

### PR DESCRIPTION
**Pull request by jackkleeman**
i-can't-know-how-to-hear-anymore-about-tables-i-think-you-should-leave-with-tim-robinson

Also add service_key_json cols

Best to review without whitespace, as i formatted the existing reference tables

### CODE REVIEW

**Review by slinkydeveloper — COMMENTED** [review]
I think you also need to add these new fields https://github.com/restatedev/restate/pull/913

File: docs/references/restate-sql-introspection.mdx [hunk 1]
```diff
@@ -10,35 +10,75 @@ The schema of these tables is described in the following.
...
+| `invoked_by`         | `Utf8`   | Enum describing the invoker of this service: `ingress` = invoked externally / `service` = invoked by a service.                                                                                                                                              |                                    |
+| `invoked_by_service` | `Utf8`   | The name of the invoking service. Or `null` if invoked via the ingress.                                                                                                                                                                                      | `com.example.Greeter`              |
+| `invoked_by_id`      | `Utf8`   | The Invocation ID of the service that triggered this invocation. Or `null` if invoked externally.                                                                                                                                                            |                                    |
```
Comment (tillrohrmann) [inline docs/references/restate-sql-introspection.mdx:42]: `pinned_endpoint_id` seems to be missing.

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @jackkleeman. There seem to be a few fields missing that were added lately.

File: docs/references/restate-sql-introspection.mdx [hunk 2]
```diff
@@ -10,35 +10,75 @@ The schema of these tables is described in the following.
...
+| `retry_count`       | `UInt64`  | The number of attempts for this particular execution of this invocation. Increments on start, so 2 or more means a failure occurred.                                                                                                                         |                                    |
+| `last_start_at`     | `Date64`  | Timestamp indicating the start of the most recent execution of this invocation.                                                                                                                                                                              |                                    |
+| `last_failure`      | `Utf8`    | An error message describing the most recent failure to execute this invocation, if any.                                                                                                                                                                      |                                    |
```
[the remark this answers is not in the document]
Comment (jackkleeman) [reply docs/references/restate-sql-introspection.mdx]: can you refresh? i added these already

File: docs/references/restate-sql-introspection.mdx (hunk 1 above)
Comment (jackkleeman) [reply docs/references/restate-sql-introspection.mdx:42]: can you refresh? i added this already

**Pull request merged**